File: 027.Remove Element/027.Remove Element.java

//----------------java--------------------
public class Solution {
    public int removeElement(int[] A, int elem) {
        //{1,1,2,1,2,3,2,3,1}
        int newIndex = 0;
        for (int oldIndex = 0; oldIndex < A.length; ++oldIndex) {
            if (A[oldIndex] != elem) {
                A[newIndex++] = A[oldIndex];
            } 
        }
        return newIndex;
    }
}

File: 027.Remove Element/027.Remove Element.cpp
//----------cpp------------
class Solution {
	public:
		int removeElement(vector<int>& nums, int val) {
			int n = nums.size(), p = 0;
			for (int i = 0; i < n; i++) {
				if (val != nums[i])
					nums[p++] = nums[i];
			}
			return p;
		}
};
